implementation/servlet/src/main/java/net/opentsdb/servlet/resources/RawQueryRpc.java

```diff
@@ -94,6 +113,34 @@ public Response post(final @Context ServletConfig servlet_config,
           .build();
     }
     
+    return handleQueryAsync(servlet_config, request);
+  }
+  
+  private Response handleQueryAsync(ServletConfig servlet_config, HttpServletRequest request) throws Exception {
+
+    TSDQueryMeta qm = new TSDQueryMeta(servlet_config, request, false);
+
+    Object obj = servlet_config.getServletContext().getAttribute(OpenTSDBApplication.TSD_ATTRIBUTE);
+    if (obj == null) {
+      throw new WebApplicationException("Unable to pull TSDB instance from " + "servlet context.",
+          Response.Status.INTERNAL_SERVER_ERROR);
+    } else if (!(obj instanceof TSDB)) {
+      throw new WebApplicationException(
+          "Object stored for as the TSDB was " + "of the wrong type: " + obj.getClass(),
+          Response.Status.INTERNAL_SERVER_ERROR);
+    }
+    final TSDB tsdb = (TSDB) obj;
+
+    RunTSDQuery runTsdQuery = new RunTSDQuery(qm);
+    
+    LOG.info("Creating async query");
```

Comment:
Reason why it made sense for me to parse in async, is we don't want to decrement the counter in the Query registration listener when the query fails before it gets scheduled. And when a query is bad (syntax or auth), it would fail much quicker in the threadpool so the other valid queries would be unblocked soon.

Reply:
We could register the queries after they've been constructed and planned, that way the threadpool only handles the queries that *would* execute and the bad queries will be handled in the Undertow threads.